core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -353,6 +354,25 @@ class SparkHadoopUtil extends Logging {
     }
     buffer.toString
   }
+
+  /**
+   * Run some code as the real logged in user (which may differ from the current user, for
+   * example, when using proxying).
+   */
+  private[spark] def doAsRealUser[T](fn: => T): T = {
+    val currentUser = UserGroupInformation.getCurrentUser()
```

Comment:
I see. Let me take a try. But I'm guessing this is the only place where the issue can be handled from Spark side.